sdk/storage/storage-blob/src/policies/RetryPolicy.ts

```diff
@@ -1,338 +0,0 @@
-   * @param {HttpOperationResponse} response
-   * @param {boolean} secondaryHas404  If attempt was against the secondary & it returned a StatusNotFound (404), then
-   *                                   the resource was not found. This may be due to replication delay. So, in this
-   *                                   case, we'll never try the secondary again for this operation.
-   * @param {number} attempt           How many retries has been attempted to performed, starting from 1, which includes
-   *                                   the attempt will be performed by this method call.
-   * @returns {Promise<HttpOperationResponse>}
-   * @memberof RetryPolicy
-   */
-  protected async attemptSendRequest(
-    request: WebResource,
-    secondaryHas404: boolean,
-    attempt: number
-  ): Promise<HttpOperationResponse> {
-    const newRequest: WebResource = request.clone();
-
-    const isPrimaryRetry =
-      secondaryHas404 ||
-      !this.retryOptions.secondaryHost ||
-      !(request.method === "GET" || request.method === "HEAD" || request.method === "OPTIONS") ||
-      attempt % 2 === 1;
-
-    if (!isPrimaryRetry) {
-      newRequest.url = setURLHost(newRequest.url, this.retryOptions.secondaryHost!);
-    }
-
-    // Set the server-side timeout query parameter "timeout=[seconds]"
-    if (this.retryOptions.tryTimeoutInMs) {
-      newRequest.url = setURLParameter(
-        newRequest.url,
-        URLConstants.Parameters.TIMEOUT,
-        Math.floor(this.retryOptions.tryTimeoutInMs! / 1000).toString()
-      );
-    }
-
-    let response: HttpOperationResponse | undefined;
-    try {
-      this.logf(
-        HttpPipelineLogLevel.INFO,
-        `RetryPolicy: =====> Try=${attempt} ${isPrimaryRetry ? "Primary" : "Secondary"}`
-      );
-      response = await this._nextPolicy.sendRequest(newRequest);
-      if (!this.shouldRetry(isPrimaryRetry, attempt, response)) {
-        return response;
-      }
-
-      secondaryHas404 = secondaryHas404 || (!isPrimaryRetry && response.status === 404);
-    } catch (err) {
-      this.logf(
-        HttpPipelineLogLevel.ERROR,
-        `RetryPolicy: Caught error, message: ${err.message}, code: ${err.code}`
-      );
-      if (!this.shouldRetry(isPrimaryRetry, attempt, response, err)) {
-        throw err;
-      }
-    }
-
-    await this.delay(isPrimaryRetry, attempt, request.abortSignal);
-    return await this.attemptSendRequest(request, secondaryHas404, ++attempt);
-  }
-
-  /**
-   * Decide whether to retry according to last HTTP response and retry counters.
-   *
-   * @protected
-   * @param {boolean} isPrimaryRetry
-   * @param {number} attempt
-   * @param {HttpOperationResponse} [response]
-   * @param {RestError} [err]
-   * @returns {boolean}
-   * @memberof RetryPolicy
-   */
-  protected shouldRetry(
-    isPrimaryRetry: boolean,
-    attempt: number,
-    response?: HttpOperationResponse,
-    err?: RestError
-  ): boolean {
-    if (attempt >= this.retryOptions.maxTries!) {
```

Comment:
Different services may have different retry polices? Does RetryPolicy has same policies as storage's one. Or where we define the specific retry logic as we do here?

Reply:
Agree. Retry policy includes storage specific retry logics, not sure this can be applied for other SDKs.

Reply:
Yeah, I think we have to bring back Storage's custom retry policy and make this pluggable.

Reply:
Brought this back, along with StoragePipelineOptions that extends PipelineOptions.